### ISSUE TITLE
Issues running on Raspberry Pi

### ISSUE THREAD

**Issue by jamalex** [learningequality/kolibri-server#10]
Installing on a Raspberry Pi running Raspbian. Tried to use the PPA, but got:

```
$ sudo add-apt-repository ppa:learningequality/kolibri-proposed

Traceback (most recent call last):
  File "/usr/bin/add-apt-repository", line 95, in <module>
    sp = SoftwareProperties(options=options)
  File "/usr/lib/python3/dist-packages/softwareproperties/SoftwareProperties.py", line 109, in __init__
    self.reload_sourceslist()
  File "/usr/lib/python3/dist-packages/softwareproperties/SoftwareProperties.py", line 599, in reload_sourceslist
    self.distro.get_sources(self.sourceslist)    
  File "/usr/lib/python3/dist-packages/aptsources/distro.py", line 89, in get_sources
    (self.id, self.codename))
aptsources.distro.NoDistroTemplateException: Error: could not find a distribution template for Raspbian/stretch
```

So then I downloaded the DEB's and installed them with `dpkg`. Everything worked until it tried to start the server. Starting the server manually gave:

```
$ sudo service kolibri-server start
Job for kolibri-server.service failed because a timeout was exceeded.
See "systemctl status kolibri-server.service" and "journalctl -xe" for details.
```

```
systemctl status kolibri-server.service

● kolibri-server.service - A high performance web server setup for Kolibri
   Loaded: loaded (/lib/systemd/system/kolibri-server.service; enabled; vendor preset: enabled)
   Active: failed (Result: timeout) since Thu 2019-03-21 15:13:55 PDT; 23s ago
  Process: 20877 ExecStart=/etc/init.d/kolibri-server start (code=killed, signal=TERM)
   CGroup: /system.slice/kolibri-server.service
           └─20928 systemctl restart kolibri.service

Mar 21 15:12:25 kolibri-bash su[20889]: Successful su for pi by root
Mar 21 15:12:25 kolibri-bash su[20889]: + ??? root:pi
Mar 21 15:12:25 kolibri-bash su[20889]: pam_unix(su:session): session opened for user pi by (uid=0)
Mar 21 15:12:28 kolibri-bash kolibri-server[20877]: WARNING:root:No C Extensions available for this platform.
Mar 21 15:12:34 kolibri-bash kolibri-server[20877]: WARNING  Options file /home/pi/.kolibri/options.ini has been updated; server restart is required before change will take effect.
Mar 21 15:12:34 kolibri-bash kolibri-server[20877]: debconf: DbDriver "passwords" warning: could not open /var/cache/debconf/passwords.dat: Permission denied
Mar 21 15:13:55 kolibri-bash systemd[1]: kolibri-server.service: Start operation timed out. Terminating.
Mar 21 15:13:55 kolibri-bash systemd[1]: Failed to start A high performance web server setup for Kolibri.
Mar 21 15:13:55 kolibri-bash systemd[1]: kolibri-server.service: Unit entered failed state.
Mar 21 15:13:55 kolibri-bash systemd[1]: kolibri-server.service: Failed with result 'timeout'.
```

```
$ journalctl -xe

Mar 21 15:19:50 kolibri-bash kolibri[21260]: WARNING:root:No C Extensions available for this platform.
Mar 21 15:20:00 kolibri-bash kolibri[21260]: INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
Mar 21 15:20:45 kolibri-bash systemd[1]: kolibri-server.service: Start operation timed out. Terminating.
Mar 21 15:20:45 kolibri-bash systemd[1]: Failed to start A high performance web server setup for Kolibri.
-- Subject: Unit kolibri-server.service has failed
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- Unit kolibri-server.service has failed.
-- 
-- The result is failed.
Mar 21 15:20:45 kolibri-bash systemd[1]: kolibri-server.service: Unit entered failed state.
Mar 21 15:20:45 kolibri-bash systemd[1]: kolibri-server.service: Failed with result 'timeout'.
Mar 21 15:20:45 kolibri-bash sudo[21115]: pam_unix(sudo:session): session closed for user root
Mar 21 15:20:48 kolibri-bash kolibri[21260]: INFO     Sqlite database Vacuum finished.
Mar 21 15:20:48 kolibri-bash kolibri[21260]: INFO     Running update routines for new version...
Mar 21 15:20:49 kolibri-bash kolibri[21260]: 0 static files copied to '/home/pi/.kolibri/static', 537 unmodified.
Mar 21 15:20:50 kolibri-bash kolibri[21260]: Operations to perform:
Mar 21 15:20:50 kolibri-bash kolibri[21260]:   Apply all migrations: admin, analytics, auth, content, contenttypes, device, discovery, exams, kolibriauth, lessons, logger, morango, notifications, sessions
Mar 21 15:20:51 kolibri-bash kolibri[21260]: Running migrations:
Mar 21 15:20:51 kolibri-bash kolibri[21260]:   No migrations to apply.
Mar 21 15:20:54 kolibri-bash kolibri[21260]: Operations to perform:
Mar 21 15:20:54 kolibri-bash kolibri[21260]:   Apply all migrations: admin, analytics, auth, content, contenttypes, device, discovery, exams, kolibriauth, lessons, logger, morango, notifications, sessions
Mar 21 15:20:54 kolibri-bash kolibri[21260]: Running migrations:
Mar 21 15:20:54 kolibri-bash kolibri[21260]:   No migrations to apply.
Mar 21 15:20:58 kolibri-bash kolibri[21260]: Installed 2 object(s) from 1 fixture(s)
Mar 21 15:20:58 kolibri-bash kolibri[21260]: INFO     Running 'kolibri start' as daemon (system service)
Mar 21 15:20:58 kolibri-bash kolibri[21260]: INFO     Starting Kolibri background services
Mar 21 15:20:58 kolibri-bash kolibri[21260]: INFO     Going to daemon mode, logging to /home/pi/.kolibri/server.log
Mar 21 15:21:00 kolibri-bash su[21271]: pam_unix(su:session): session closed for user pi
Mar 21 15:21:00 kolibri-bash systemd[1]: Started LSB: kolibri daemon, an offline education platform.
-- Subject: Unit kolibri.service has finished start-up
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- Unit kolibri.service has finished starting up.
-- 
-- The start-up result is done.
```

**Comment by jredrejo** [comment]
Hello @jamalex , I think the package is not installed properly. There are not official docs yet, so it's logical.

1. To install it using apt, the steps neededs are those written in 
https://kolibri.readthedocs.io/en/latest/install.html#rpi
replacing 
`deb http://ppa.launchpad.net/learningequality/kolibri/ubuntu xenial main`
by
`deb http://ppa.launchpad.net/learningequality/kolibri-proposed/ubuntu xenial main`

The ppa is in Ubuntu, and raspbian/stretch is a Debian version, there's no way to create a Debian repository in the ppa.


2. If the package is installed using dpkg, a second step is needed later to get all the needed dependencies and let debconf finish:
`apt-get -f install`

**Comment by indirectlylit** [comment]
Adding 0.1.0 milestone.

After some testing, @jredrejo reports:

> `kolibri-server` is running correctly - I think it was fixed rebooting the Rpi
> 
> @jamalex do you remember anything else so I can try to reproduce the problem in my RPi? I had tested it having a previous nginx config running and kolibri-server restarts it, so this is not the case.
>
> Did you have by change uwsgi running too before installing `kolibri-server`? I had not taken that case into account

**Comment by jamalex** [comment]
I didn't run uwsgi manually at all.

I stopped nginx, moved the old config from sites-enabled to sites-available, then installed the `kolibri` deb and other dependencies (which would have started the `kolibri` service) before installing `kolibri-server` deb. When it didn't work, I did try stopping the `kolibri` service manually (though I would have thought `kolibri-server` would do this too) before trying to start `kolibri-server` again.
